opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -65,10 +69,91 @@ struct ConstructionTraits<Opm::ParallelOverlappingILU0<Matrix,Domain,Range,Paral
 
 } // end namespace Amg
 
+
+
 } // end namespace Dune
 
 namespace Opm
 {
+    namespace detail
+    {
+      //! compute ILU decomposition of A. A is overwritten by its decomposition
+      template<class M, class CRS, class InvVector>
+      void convertToCRS(const M& A, CRS& lower, CRS& upper, InvVector& inv )
+      {
+        typedef typename M :: size_type size_type;
+
+        // count non-zeros in case 0 is returned
+        // this is due to a bug in dune-istl < 2.5
+        size_type nonZeros = A.nonzeroes();
+        if( nonZeros == 0 )
+        {
+            // count non-zeros
+            const auto end = A.end();
+            for( auto row = A.begin(); row != end; ++row )
+            {
+                nonZeros += (*row).size();
+            }
+        }
+
+        lower.resize( A.N(), nonZeros );
+        upper.resize( A.N(), nonZeros );
```

Comment:
As long as we do not add any matrix entries resembling connections due to well perforations, the sparsity pattern should be perfectly symmetric. For the other case as safety factor (1.3?) should suffice in most cases.

Would `resize((nonZeros+1/2)*1.3)` also work? That would save us the space of half a matrix in the meantime.